### PR TITLE
port/mimxrt: Add MIMXRT1064 board

### PR DESCRIPTION
**Pull request by Albort12138**

### CODE REVIEW

**Comment by yanyao12138** [comment]
Hi @jimmo please help me review this PR. Thank you very much.

File: ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MCU_SERIES = MIMXRT1064
+MCU_VARIANT = MIMXRT1064DVL6A
+
+JLINK_PATH = /media/RT1064-EVK/
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk:4]: other boards have this as `JLINK_PATH ?= ...`

File: ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+CFLAGS += -DBOARD_FLASH_SIZE=0x400000
```
Comment (dpgeorge) [inline ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk:6]: other boards have this as a `#define` in the `mpconfigboard.h` file; can the same be done for this board?

File: ports/mimxrt/Makefile [hunk 3]
```diff
@@ -40,7 +40,7 @@ INC += -I$(TOP)/lib/tinyusb/hw
 INC += -I$(TOP)/lib/tinyusb/hw/bsp/teensy_40
 
 CFLAGS_MCU = -mtune=cortex-m7 -mcpu=cortex-m7 -mfloat-abi=hard -mfpu=fpv5-d16
-CFLAGS = $(INC) -Wall -Werror -Wdouble-promotion -Wfloat-conversion -std=c99 -nostdlib -mthumb $(CFLAGS_MCU)
+CFLAGS += $(INC) -Wall -Werror -Wdouble-promotion -Wfloat-conversion -std=c99 -nostdlib -mthumb $(CFLAGS_MCU)
```
Comment (dpgeorge) [inline ports/mimxrt/Makefile:43]: this change is not needed if `BOARD_FLASH_SIZE` can be changed to a `#define`

File: ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk (hunk 2 above)
Comment (yanyao12138) [reply ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk:6]: The macro *BOARD_FLASH_SIZE* is used in this [file](https://github.com/hathach/nxp_driver/blob/b618cb1d521cc9e133bdcd0fca154dee2d925dfe/sdk/devices/MIMXRT1064/xip/fsl_flexspi_nor_boot.h), and this file only includes "board.h". I think only the definition in "board.h" is useful and the definition of "BOARD_FLASH_SIZE" in mpconfigboard.h is useless. But in currently nxp_driver submodule, BOARD_FLASH_SIZE is not defined in [board.h](https://github.com/hathach/nxp_driver/blob/b618cb1d521cc9e133bdcd0fca154dee2d925dfe/sdk/devices/MIMXRT1064/project_template/board.h). So for this boards, I use this way to define BOARD_FLASH_SIZE macro.

File: ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk (hunk 1 above)
Comment (yanyao12138) [reply ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk:4]: Will update

File: ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk (hunk 2 above)
Comment (dpgeorge) [reply ports/mimxrt/boards/MIMXRT1064_EVK/mpconfigboard.mk:6]: ok, sounds good then!

**Comment by dpgeorge** [comment]
Thanks for updating, merged in 5f50568b1f7ab4f917b8d782d643ac55612910dd

**Pull request closed without merging**